src/coreComponents/mesh/CellElementRegion.cpp

```diff
@@ -17,16 +17,54 @@
 #include "CellElementSubRegion.hpp"
 #include "mesh/generators/CellBlockABC.hpp"
 
+#include <fnmatch.h>
+
 namespace geos
 {
 using namespace dataRepository;
 
 CellElementRegion::CellElementRegion( string const & name, Group * const parent ):
   ElementRegionBase( name, parent )
 {
+  std::vector< string > elementNames;
+  for( int i = 0; i < numElementTypes(); ++i )
+  {
+    if( getElementDim( (ElementType)i ) == 3 )
+    {
+      elementNames.push_back( getElementTypeName( (ElementType)i ) );
+    }
+  }
+
   registerWrapper( viewKeyStruct::sourceCellBlockNamesString(), &m_cellBlockNames ).
     setRTTypeName( rtTypes::CustomTypes::groupNameRefArray ).
-    setInputFlag( InputFlags::REQUIRED );
+    setInputFlag( InputFlags::OPTIONAL ).
+    setDescription( GEOS_FMT( "List of the desired cell-blocks names to contain in this {}.\n"
+                              "The form of this attribute is of \"regionAttribute_elementType\", so \"1_tetrahedra\" select the "
+                              "cell-block that contains the tetrahedric elements for which the regionAttribute is 1.\n"
+                              "The element types are: {}.\n"
+                              "This setting cannot be used simultaneously with {} nor {}.",
+                              catalogName(), stringutilities::join( elementNames, ", " ),
+                              viewKeyStruct::cellBlockAttributeValuesString(),
+                              viewKeyStruct::cellBlockMatchPatternsString() ) );
+
+  registerWrapper( viewKeyStruct::cellBlockAttributeValuesString(), &m_cellBlockAttributeValues ).
+    setInputFlag( InputFlags::OPTIONAL ).
+    setDescription( GEOS_FMT( "List of regionAttribute values for which we want to add the cells in this {}.\n"
+                              "I.e. {{ 1, 2 }} selects the {{ 1_tetrahedra, 1_pyramid, 2_tetrahedra, 2_pyramid... }} cellBlocks.\n"
+                              "This setting cannot be used simultaneously with {} nor {}.",
+                              catalogName(),
+                              viewKeyStruct::sourceCellBlockNamesString(),
+                              viewKeyStruct::cellBlockMatchPatternsString() ) );
```

Comment:
With the match pattern option, is this option redundant?

Reply:
I don't think so because it hides the cell-blocks concept, however I merged it too (see my following message).